nfs/inc/rpc_readdir.h

```diff
@@ -306,7 +306,7 @@ struct readdirectory_cache
      */
     bool is_confirmed() const;
 
-    bool add(struct directory_entry* entry, bool acquire_lock = true);
+    bool add(std::shared_ptr<struct directory_entry> entry, bool acquire_lock = true);
```

Comment:
the 1st param can be const shared_ptr& as add() calls std::map emplace() which will grab a new ref on the shared_ptr